src/ansible_compat/roles.py

```diff
@@ -0,0 +1,32 @@
+"""Wrapper class for Ansible Roles."""
+from keyword import iskeyword
+
+
+class Role:
+    """Utility class for validating roles."""
```

Comment:
This is still a concern, I did not resolve it.

Reply:
I know that you prefer the functional implementation but the reason why I went for the extra complex  class based approach is that this would be really needed for validating roles as we would need extra context when creating them. Imagine that a role validity depends on the context, usually of an ee, or local ansible install. If it would be simple function that validates roles, it would easily end-up having an insane signature.

Reply:
It's just one context arg in the signature.